src/module.c

```diff
@@ -6704,6 +6721,50 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+/* Check if the command can be executed by the user created through the redis module
+ * interface, according to the ACLs associated to it.
+ *
+ * If the user can execute the command, REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned. */
+int RM_ACLCheckCommandPerm(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    int keyidxptr;
+    struct redisCommand *cmd;
+
+    /* Find command */
+    if ((cmd = lookupCommand(argv[0]->ptr)) == NULL) {
+        errno = ENOENT;
+        return REDISMODULE_ERR;
+    }
+
+    if (ACLCheckAllUserCommandPerm(ctx->client->user, cmd, argv, argc, &keyidxptr) != ACL_OK) {
+        errno = EACCES;
+        return REDISMODULE_ERR;
+    }
+
+    return REDISMODULE_OK;
+}
+
+/* Check if the key can be accessed by the user created through the redis module
+ * interface, according to the ACLs associated to it.
+ *
+ * If the user can access the key, REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned. */
+int RM_ACLCheckKeyPerm(RedisModuleCtx *ctx, const char *key, int keylen) {
+    return ACLCheckKey(ctx->client->user, key, keylen) == ACL_OK ? REDISMODULE_OK : REDISMODULE_ERR;
+}
+
+/* Check if the channel can be accessed by the user created through the redis module
+ * interface, according to the ACLs associated to it.
+ *
+ * If the user can access the key, REDISMODULE_OK is returned, otherwise
+ * REDISMODULE_ERR is returned. */
+int RM_ACLCheckChannelPerm(RedisModuleCtx *ctx, const char *ch, int chlen, int literal) {
```

Comment:
I think the input should be a RedisModuleString, and we also need to document the `literal` flag.